File: FlashCards - ruby/FlashCards/cards.rb
class Cards

    @@filepath = nil
    @@filecat = File.join(APP_ROOT, 'files', 'categories')

    def self.filepath=(path=nil)
      @@filepath = File.join(APP_ROOT, 'files', path)
    end

    attr_accessor :front, :back
    
    def self.file_exists?
      if @@filepath && File.exists?(@@filepath)
        return true
      else
        return false
      end
    end
    
    def self.file_usable?
      return false unless @@filepath 
      return false unless File.exists?(@@filepath)
      return false unless File.readable?(@@filepath)
      return false unless File.writable?(@@filepath)
      return true
    end

    def self.cat_file_usable?
        return false unless @@filecat
        return false unless File.exists?(@@filecat)
        return false unless File.readable?(@@filecat)
        return true
      end
    
    def self.create_file
      File.open(@@filepath, 'w') unless file_exists?
      return file_usable?
    end
    
    def save_card
        return false unless Cards.file_usable?
        File.open(@@filepath, 'a') do |file|
          file.puts "#{[@front, @back].join("\t")}\n"
        end
        return true
    end

    def self.saved_categories
        categories = []
        if cat_file_usable?
            file = File.new(@@filecat, 'r')
            file.each_line do |line|
                categories.push(line)
            end
            file.close
        end
        return categories
    end

    def self.save_category_name(category)
      File.open(@@filecat, 'a') do |file|
        file.puts "#{category}\n"
      end
    end

    def self.saved_card
        cards = []
        if file_usable?
            file = File.new(@@filepath, 'r')
            file.each_line do |line|
                cards << Cards.new.import_line(line.chomp)
            end
            file.close
        end
        
        return cards.shuffle
    end

    def import_line(line)
        line_array = line.split("\t")
        @front = line_array[0]
        @back = line_array[1]
        return self
    end
    
end

File: FlashCards - py/FlashCards/cards.py
import sys
import random

class Cards :
    path = ""

    def __init__(self, front, back) :
        self.front = front
        self.back = back

    def make_category_file(self) :
        file = open("files/categories.txt", "a")
        file.close
        return True

    def save_category_name(self, category) :
        file = open("files/categories.txt", "a")
        file.write(str(category) + "\n")
        file.close

        path = str(category.strip)
        cat_file = open("files/" + category + ".txt", "a")

    def saved_categories(self) :
        categories = []
        file = open("files/categories.txt", "r")
        for category in file:
            categories.append(category.strip())
        file.close
        return categories
    
    def save_card(self, category, card) :
        path = str(category.strip())
        file = open("files/" + path + ".txt", "a")
        file.write(str(card) + "\n")
        file.close
        return True 

    def saved_card(self, path) :
        cardsList = []
        p = str(path.strip())
        file = open("files/" + p + ".txt", "r")
        for cards in file:
            make = cards.strip().split("\t")
            front = make[0]
            back = make[1]
            nwCard = Cards(front, back) 
            cardsList.append(nwCard)

        file.close
        random.shuffle(cardsList)
        return cardsList   

    # def import_line(line) :
    #     line_array = line.split("\t") 
    #     self.front = line_array[0]
    #     self.back = line_array[1]
    #     return self


        